internal/controller/eventing/nats_test.go

```diff
@@ -138,17 +162,47 @@ func Test_reconcileNATSSubManager(t *testing.T) {
 			wantAssertCheck:  true,
 			givenShouldRetry: true,
 			wantError:        errors.New("failed to start"),
+			wantHashAfter:    uint64(0),
+		},
+		{
+			name:                         "it should update the subscription manager when the backend config changes",
+			givenIsNATSSubManagerStarted: false,
+			givenHashBefore:              uint64(0),
```

Comment:
should it not have a hash before to see if existing has changes to a new one?

Reply:
The test used to go through the whole flow: empty subManager -> gets a hash -> then update the manager -> new hash. But as the case of the new subManager is already covered, I changed this to not empty subManager -> update config -> new hash.